Edugate/Jagger
# support for OIDC (experimental)

- some code cleaning

_support for OIDC (experimental)_
- should work with OpenidConnect Providers if they:
  - follow specification
  - provide .well-known/openid-configuration  
  - provide jkws-uri
  - id_token is signed with alg RS256
  - support Authorization Code Flow
- tested with Google

_how to enable it in jagger_
- download https://github.com/janul/jagger-oidc
- copy Jagger folder from jagger-oidc into {JAGGERAPP}/application/extplugins/   _(this is temporary way)_
- in config_rr.php
  - enable oidc:
    - **$config['oidc_enabled'] = true;**
  - list of OPs:
    - **$config['oidc_ops'] = array()**
  - example for google:
    - $config['oidc_ops'] = array(
      
      'https://accounts.google.com' => array(
           'name'=>'Google',
            'client_id'=>'CLIENT ID',
           'client_secret'=>'CLIENT SECRET',
           'scopes'=>array('email','profile'),
           'authzparams' => array( 'prompt'=>'select_account' ),
           'openid_configuration' => array(),
           'mapping_claims' => array(
               'fname'=>'given_name',
               'sname'=>'family_name',
              'email'=>'email'
        ))); 
- additional button should appear on login modal form _(improving rempleate in progress)_
- username value is created based on 'sub' claim and 'iss' : 'sub'@'iss'    and access type is set as federated
